uni/lib/view/library/widgets/library_occupation_tab.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:flutter/material.dart';
+import 'package:percent_indicator/linear_percent_indicator.dart';
+import 'package:provider/provider.dart';
+import 'package:uni/model/entities/library_occupation.dart';
+import 'package:uni/model/providers/library_occupation_provider.dart';
+import 'package:uni/model/request_status.dart';
+import 'package:uni/view/common_widgets/page_title.dart';
+import 'package:uni/view/library/widgets/library_occupation_card.dart';
+
+class LibraryOccupationTab extends StatefulWidget {
+  const LibraryOccupationTab({Key? key}) : super(key: key);
+
+  @override
+  LibraryOccupationTabState createState() => LibraryOccupationTabState();
+}
+
+class LibraryOccupationTabState extends State<LibraryOccupationTab> {
+  @override
+  Widget build(BuildContext context) {
+    return Consumer<LibraryOccupationProvider>(
+        builder: (context, occupationProvider, _) {
+      if (occupationProvider.status == RequestStatus.busy) {
+        return const Center(child: CircularProgressIndicator());
+      } else {
+        return LibraryOccupationTabView(occupationProvider.occupation);
+      }
+    });
+  }
+}
+
+class LibraryOccupationTabView extends StatelessWidget {
+  final LibraryOccupation? occupation;
+
+  const LibraryOccupationTabView(this.occupation, {super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    if (occupation == null || occupation?.capacity == 0) {
+      return ListView(scrollDirection: Axis.vertical, children: [
+        Center(
+            heightFactor: 2,
+            child: Text('Não existem dados para apresentar',
+                style: Theme.of(context).textTheme.headline6,
+                textAlign: TextAlign.center))
+      ]);
+    }
+    return ListView(
+        scrollDirection: Axis.vertical,
+        shrinkWrap: true,
+        children: [
+          LibraryOccupationCard(),
+          if (occupation != null) const PageTitle(name: 'Pisos'),
+          if (occupation != null) FloorRows(occupation!),
```

Comment:
I think this works too

```suggestion
          if(occupation != null) ...[
              const PageTitle(name: 'Pisos'),
              FloorRows(occupation!),
          ],
   
```